### PR TITLE
[SDESK-346] - Related item widget improvements

### PR DESCRIPTION
**Pull request by akintolga**
- If the item has existing links, the widget will display them
- If the item has no links then the widget will display `linkable` stories

### CODE REVIEW

File: scripts/apps/archive/services/FamilyService.js [hunk 1]
```diff
@@ -13,20 +31,150 @@ export function FamilyService(api, desks) {
...
+     * @public
+     * @description Returns any story potentially linkable
+     * @param {string} keyword - slugline to be matched
+     * @param {string} sluglineMatch - type of matching rule for slugline
+     * @param {string} eventId - event id to be searched
+     * @param {string} modificationDateAfter - filter for versioncreated
+     * @returns {Object}
+     */
+    this.fetchRelatableItems = (keyword, sluglineMatch, eventId, modificationDateAfter) => {
+        var filter = [
```
Comment (mdhaman) [inline scripts/apps/archive/services/FamilyService.js]: Do we need to add filter for the `last_published_version` for the items from the published collection?

Comment (akintolga) [inline scripts/apps/archive/services/FamilyService.js]: That's true. I was hiding the non-last versions on the view but it is better if I added here as well.

Comment (akintolga) [inline scripts/apps/archive/services/FamilyService.js]: Done.

File: scripts/apps/archive/related-item-widget/relatedItem.js [hunk 2]
```diff
@@ -70,6 +73,25 @@ function RelatedItemController(
         sort: [{versioncreated: 'desc'}]
     };
 
+    $scope.loading = true;
+    familyService.fetchRelatedItems($scope.item.event_id)
```
Comment (petrjasek) [inline scripts/apps/archive/related-item-widget/relatedItem.js]: imo would be better to have `item` as param to `fetchRelatedItems` and hide implementation that it's using `event_id` field

Comment (akintolga) [inline scripts/apps/archive/related-item-widget/relatedItem.js]: Done.

**Pull request merged**